linker/shared/src/main/scala/org/scalajs/linker/irio/MemOutputFileImpl.scala

```diff
@@ -18,22 +18,26 @@ import java.io._
 import java.nio.ByteBuffer
 import java.nio.charset.StandardCharsets
 
-final class WritableMemVirtualBinaryFile extends WritableVirtualBinaryFile {
+import org.scalajs.linker.LinkerOutput
+
+private[linker] final class MemOutputFileImpl extends LinkerOutput.MemFile {
   private var _content: Array[Byte] = _
 
   def content: Array[Byte] = _content
 
-  def newChannel()(implicit ec: ExecutionContext): Future[WriteChannel] =
-    Future.successful(new Channel)
+  private[linker] object impl extends OutputFile {
```

Comment:
Then we would expose a less stable type in the stable interface. Because `OutputFile` has all the accessor methods. So this design indeed does not allow to define a type that:
- implements `LinkerOutput.File`
- provides additional API.
- does not expose its implementation.

This is fixable, but probably requires even more types :)

Reply:
Seems the only way to fix this is a sealed trait. I'll do that.